### ISSUE TITLE
[Bug]: The Bottles runtime is broken in several ways

### ISSUE THREAD

**Issue by koplo199** [bottlesdevs/Bottles#3054]
It is not possible to enable the Bottles runtime using the GUI in flatpak (nor it is to update it) even though it is included in the [build manifest](https://github.com/bottlesdevs/Bottles/blob/44b06047c87af2ed3620c3fa8dcd10a63227b247/com.usebottles.bottles.yml#L265-L280). It can be only enabled by manually flipping `use_runtime` to `true` in a bottle config file.

For users it means the following: 
1. Not benefiting from compatibility/missing libraries included in the Bottles runtime, unless the user know it can be enabled by manually modifying their bottles configuration files.
2. Possibly using an outdated Bottles runtime until a release is made, which can take months.

There is also no way of telling Bottles to prefer system libraries, in order to use the Bottles runtime purely for missing libraries.

But currently it doesn't matter, as enabling the Bottles runtime (alone) is pure placebo.

This is the value concatenated to the `LD_LIBRARY_PATH` env variable:
`:::/:a:p:p:/:e:t:c:/:r:u:n:t:i:m:e:/:l:i:b:::/:a:p:p:/:e:t:c:/:r:u:n:t:i:m:e:/:l:i:b:3:2:::/:a:p:p:/:l:i:b:::/:a:p:p:/:l:i:b:3:2`
Where it was suposed to be:
`:/app/etc/runtime/lib:/app/etc/runtime/lib32:/app/lib:/app/lib32`

<details>
  This is caused by this line:

https://github.com/bottlesdevs/Bottles/blob/0914b2e34c1800784b26ca52d1a80f06d65252fa/bottles/backend/wine/winecommand.py#L218

  Which should be instead:

  ```Python
  ld.append(_rb)
  ```
</details>

But, even if it was set to the value above, it would still be broken as the path is incorrect. Looking at a [Bottles runtime archive](https://github.com/bottlesdevs/runtime/releases/tag/0.6.3), 64/32bit libraries are copied inside `/app/etc/runtime/lib/x86_64-linux-gnu` and `/app/etc/runtime/lib/i386-linux-gnu` respectively. 

Setting those values, and fixing other minor bugs along the way, allow programs to use the Bottles runtime libraries. 

Note: The Bottles runtime and the system libraries (the Gnome runtime in the case of flatpak) currently have priorities over runners libraries, which probably shouldn't be the case.

Even when fixed using the above steps, the Bottles runtime is missing some libraries needed to make `wine-ge` runners operate without errors:
```
(wine:157): GStreamer-WARNING **: 10:10:48.024: Failed to load plugin '/home/user/.var/app/com.usebottles.bottles/data/bottles/runners/wine-ge-proton8-13/lib64/gstreamer-1.0/libgstvpx.so': libvpx.so.5: cannot open shared object file: No such file or directory
(wine:157): GStreamer-WARNING **: 10:10:48.024: Failed to load plugin '/home/user/.var/app/com.usebottles.bottles/data/bottles/runners/wine-ge-proton8-13/lib64/gstreamer-1.0/libgstdtls.so': libcrypto.so.1.1: cannot open shared object file: No such file or directory
(wine:157): GStreamer-WARNING **: 10:10:48.025: Failed to load plugin '/home/user/.var/app/com.usebottles.bottles/data/bottles/runners/wine-ge-proton8-13/lib64/gstreamer-1.0/libgstsoup.so': libsoup-2.4.so.1: cannot open shared object file: No such file or directory
(wine:157): GStreamer-WARNING **: 10:10:48.025: Failed to load plugin '/home/user/.var/app/com.usebottles.bottles/data/bottles/runners/wine-ge-proton8-13/lib64/gstreamer-1.0/libgstwebp.so': libwebp.so.6: cannot open shared object file: No such file or directory
```
But as there is a chance that those libraries version will change rather soon™ (https://github.com/lutris/buildbot/issues/175), this last error is not worth investigating now.

**TL;DR**:

1. Bottles runtime activation broken
2. Poor Bottles runtime flatpak integration
3. Bottles runtime missing libraries for working as intended on some supported runners
4. No way of preferring system libraries over Bottles runtime libraries

**Comment by koplo199** [comment]
Note: the Lutris runtime/buildbot got updated from `Ubuntu 18.04` to `Debian 12 (Bookworm)`, and Wine-GE runners starting from the [8-16](https://github.com/GloriousEggroll/wine-ge-custom/releases/tag/GE-Proton8-16) version are using this build environment.

Libraries used for the Lutris runtime are listed [here](https://github.com/lutris/buildbot/blob/master/runtime/Debian-12.packages), porting them to the [recipe.yml](https://github.com/bottlesdevs/build-tools/blob/main/runtime/recipe.yml) used to build the Bottles runtime is trivial.

What is not trivial however, is implementing an auto-update mechanism for the Bottles runtime. Currently, even for non-Flatpak users, it is only installed once on the first launch, and then never updated again.

It is a multi-repository effort and will require a single source of truth to determine the runtime version. There is currently two of them (and out of sync): the runtime repository [version](https://github.com/bottlesdevs/runtime/blob/main/VERSION), and the build-tools runtime [version](https://github.com/bottlesdevs/build-tools/blob/main/runtime/recipe.yml#L3). As the runtime repository `VERSION` is the trigger for new runtime releases, the best solution is probably to just pass its value to the build-tools [build.sh](https://github.com/bottlesdevs/runtime/blob/07b7c22053b3c2f442f773ca9f506acb4f8a9f62/.github/workflows/build.yml#L26) script.

**Comment by koplo199** [comment]
Fixes on the [runtime](https://github.com/bottlesdevs/runtime/pull/7), [build-tools](https://github.com/bottlesdevs/build-tools/pull/6) and [runtimezilla](https://github.com/mirkobrombin/runtimezilla/pull/3) sides submitted.

**Comment by Arcitec** [comment]
Thank you for improving this situation. Sounds like I hit the same error.

I came here from Google. My Epic Games Launcher inside Bottles no longer launched. It just got stuck at a "loading progress bar" screen infinitely.

The console log showed the **exact** error messages you quoted.

I tried everything. Wine-GE 8.x, Proton-GE 8.x, Soda, Kron4ek 8.x and 9.x, etc.

The only things that worked for me were:

- Downgrade the Bottle to Wine-GE 7.41. This was the main thing that fixed it.
- Enable "Use Steam Runtime" (after using Flatseal to add `~/.var/app/com.valvesoftware.Steam` to its filesystem permissions, so that it can reach the files).

Now my Epic Games Launcher works again.

I posted these instructions to help anyone else who reaches this place via Google after their Epic Launcher breaks on Linux.

We should all migrate to Heroic Launcher though. Running the real Epic Launcher on Linux has always been tedious at best. :)
